### PR TITLE
Add support for definite integration

### PR DESCRIPTION
**Pull request by GregVernon**
See this Discourse conversation: https://discourse.julialang.org/t/integrating-symbolic-functions/102545

Thought I'd start a basic MR with the most naive implementation, and let it evolve from here.

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
>  is it enough to rely on Symbolics.substitute unit tests, and then just a few return-type / edge case tests?

It would be good to test a few values and cases.

File: src/integral.jl [hunk 1]
```diff
@@ -44,7 +46,7 @@ Output:
 - `unsolved`: the residual unsolved portion of the input
 - `err`: the numerical error in reaching the solution
 """
-function integrate(eq, x = nothing; abstol = 1e-6, num_steps = 2, num_trials = 10,
+function integrate(eq, x = nothing, domain::Vector{<:Number} = [NaN]; abstol = 1e-6, num_steps = 2, num_trials = 10,
```
Comment (ChrisRackauckas) [inline src/integral.jl:49]: shouldn't it be `[-Inf, Inf]`?

File: src/integral.jl [hunk 2]
```diff
@@ -46,7 +46,7 @@ Output:
 - `unsolved`: the residual unsolved portion of the input
 - `err`: the numerical error in reaching the solution
 """
-function integrate(eq, x = nothing, domain::Vector{<:Number} = nothing; abstol = 1e-6, num_steps = 2, num_trials = 10,
+function integrate(eq, x = nothing, domain::Vector{<:Number} = [NaN]; abstol = 1e-6, num_steps = 2, num_trials = 10,
```
Comment (ChrisRackauckas) [inline src/integral.jl:49]: for safety make it `nothing` by default.

**Comment by knuesel** [comment]
I think this can be closed as superseded by https://github.com/SciML/SymbolicNumericIntegration.jl/pull/63 ?
